main.go

```diff
@@ -97,15 +101,81 @@ func setThreads() {
 	}
 }
 
+func unixAddr(uri string) (string, error) {
+	if uri == "" {
+		return "", errors.New("empty uri")
+
+	}
+	addr, err := url.Parse(uri)
+	if err != nil {
+		return "", err
+	}
+	if addr.Scheme != "" && addr.Scheme != "unix" {
+		return "", errors.New("invalid address scheme")
+	}
+	return addr.Host + addr.Path, nil
+}
+
+func printAgentLogs(sock string) error {
+	// Don't return an error if nothing has been provided.
+	// This flag is optional.
+	if sock == "" {
+		return nil
+	}
+
+	agentLogsAddr, err := unixAddr(sock)
+	if err != nil {
+		logger().WithField("socket-address", sock).WithError(err).Fatal("invalid agent logs socket address")
+		return err
+	}
+
+	// Check permissions socket for "other" is 0.
```

Comment:
Is this a security check? If so, can you make that clear in the comment, just to make it nice and clear for the future :-)

Reply:
ok :smile: